x11-terms/st/st-0.9.2.ebuild

```
@@ -0,0 +1,75 @@
+# Copyright 1999-2024 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=8
+
+inherit desktop savedconfig toolchain-funcs
+
+DESCRIPTION="Simple terminal implementation for X"
+HOMEPAGE="https://st.suckless.org/"
+
+if [[ ${PV} == 9999 ]]; then
+	inherit git-r3
+	EGIT_REPO_URI="https://git.suckless.org/${PN}"
+else
+	SRC_URI="https://dl.suckless.org/${PN}/${P}.tar.gz"
+	KEYWORDS="~amd64 ~arm ~arm64 ~hppa ~m68k ~ppc64 ~riscv ~x86"
+fi
+
+LICENSE="MIT"
+SLOT="0"
+
+RDEPEND="
+	>=sys-libs/ncurses-6.0:0=
+	media-libs/fontconfig
+	x11-libs/libX11
+	x11-libs/libXft
+	~x11-terms/st-terminfo-${PV}
+"
+DEPEND="
+	${RDEPEND}
+	x11-base/xorg-proto
+"
+BDEPEND="virtual/pkgconfig"
+
+src_prepare() {
+	default
+
+	sed -i \
+		-e "/^X11LIB/{s:/usr/X11R6/lib:/usr/$(get_libdir)/X11:}" \
+		-e '/^STLDFLAGS/s|= .*|= $(LDFLAGS) $(LIBS)|g' \
+		-e '/^X11INC/{s:/usr/X11R6/include:/usr/include/X11:}' \
+		config.mk || die
+	sed -i \
+		-e '/tic/d' \
+		Makefile || die
+
+	restore_config config.h
+}
+
+src_configure() {
+	sed -i \
+		-e "s|pkg-config|$(tc-getPKG_CONFIG)|g" \
+		config.mk || die
+
+	tc-export CC
+}
+
+src_install() {
+	emake DESTDIR="${D}" PREFIX="${EPREFIX}"/usr install
+
+	dodoc TODO
+
+	make_desktop_entry ${PN} simpleterm utilities-terminal 'System;TerminalEmulator;' ''
+
+	save_config config.h
+}
+
+pkg_postinst() {
+	if ! [[ "${REPLACING_VERSIONS}" ]]; then
```

```suggestion
	if [[ -z "${REPLACING_VERSIONS}" ]]; then
```
How about changing this check to more traditional way used in `::gentoo`?

---

@arkamar: Thanks, I certainly can change it in the next push if needed.

According to `qgrep`, 45 ebuilds have the negated approach is in 45 ebuilds, and 159 the `-z` one.

Could you point me to some docs about this or similar traditional/preferred forms, please?

---

I am not aware of such a doc. My suggestion and impression about the form is based on similar statistic, however, I have got slightly different numbers:
```sh
git grep -- '-z.*REPLACING_VERSIONS' -- '*.ebuild' | wc -l; git grep '!.*REPLACING_VERSIONS' -- '*.ebuild' | wc -l
385
68
```
The `-z` parameter makes the condition slightly more readable in my opinion, but I understand that this is just about personal preference.

---

I plan on writing a style guide at some point, but yes, it's maintainer preference.

---

Thanks for the additional pointers! Double-checking `-z` details, and also shellcheck recommendations, I also prefer using explicit string checks.

I applied the change in a follow-up commit to all remaining x11-terms/st ebuilds after cleaning up the old versions as I did not see any reason to apply them to the ones that are going to be removed anyway.

---

Thanks! Yeah, there was no reason to apply them to the ones that are going to be removed. It would be actually enough to change this only in newly introduced version and live ebuild, because we usually don't change stable ebuilds if they work for users. However, I merged it anyway, because the change is simple enough and improves readability, in my opinion.

---

Oh, right, modifying the remaining last stable version has slipped through :sweat_smile: I'm glad it was acceptable this time!